include/smack/smack.h

```diff
@@ -58,7 +66,151 @@ int __SMACK_nondet() {
 void __SMACK_decls() {
 #define D(d) __SMACK_top_decl(d)
 
+// SMACK type definitions
+#ifdef BITVECTOR
```

Comment:
All of these type declarations (`i8`, `i16`, `i32`, `i64`, `ref`, `size`) should be output by `SmackRep::getPrelude()` since they depend on `SmackOptions::BitVectors` and the target architecture. See the methods of `llvm::DataLayout` including `getPointerTypeSize` to generate these from `SmackRep::targetData`. For generating the correct declaration for `size_t`, perhaps the parameter type to `malloc` in the bitcode could be checked.